src/main/java/Containers/Portfolio.java

```diff
@@ -23,67 +23,116 @@ public static Portfolio getInstance() {
         return Portfolio.commonPortfolio;
     }
 
-    // Storage: The UUID of asset and the reference of asset
-    private final HashMap<String, Asset> storage;
+
+
+    // assetList: a list of asset in the portfolio
+    private List<Asset> assetList;
+    // transactionList: a list of transaction in the portfolio
+    private List<Transaction> transactionList;
+    // votingHistory: a history of votes in the portfolio
+    private List<Vote> votingHistory;
+    private double profitability;
 
     private Portfolio() {
-        this.storage = new LinkedHashMap<String, Asset>();
+        this.assetList = new ArrayList<>();
+        this.transactionList = new ArrayList<>();
+        this.votingHistory = new ArrayList<>();
+        this.profitability = 0.0;
+    }
+    // Getter and Setter for profitability
+    public double getProfitability() {
+        return profitability;
+    }
+
+    public void setProfitability(double profitability) {
+        this.profitability = profitability;
+    }
+
+    // Getter and Setter for assets, transactions, and votes
```

Comment:
Your getters and setters should be near the bottom of the document by convention. Your more important methods such as adding/removing transactions/votes etc. should be near the top.